envoy/upstream/upstream.h

```diff
@@ -1172,6 +1172,11 @@ class ClusterInfo : public Http::FilterChainFactory {
    */
   virtual ClusterTimeoutBudgetStatsOptRef timeoutBudgetStats() const PURE;
 
+  /**
+   * @return true if this cluster should produce per-endpoint stats.
+   */
+  virtual bool perEndpointStats() const PURE;
```

Comment:
naming nit: call this generatePerEndpoiintStats()? Otherwise just from the name it looks like this method might be *returning* the per-endpoint stats, when looking at a call-site.

Feel free to push back of course, just offering an opinion.

Reply:
`perEndpointStatsEnabled()` seems clearest